File: lecture_notes/10 refactoring/3-step.cpp
#include <iostream>
using namespace std;

int PointsShift(const int arr[]) {

    int test[4][2] =
            {
                    {0, 0},
                    { 10, 10},
                    {20, 20},
                    {30, 30}
            };

    int polygon[4][2];

    for (auto & i : test) {
        for (int j = 0; j < 2; ++j) {
            if (j == 0) {
                cout << "x, y = " << i[j];
            } if (j == 1) {
                cout << ", " << i[j] << "\n";
            }
        }
    }
    
    cout << "\n" << "Adding new values from parameter..." << endl << "\n";
    for (auto & i : test) {
        for (int j = 0; j < 2; ++j) {
            if (j == 0) {
                cout << "x, y = " << i[j];
            } if (j == 1) {
                i[j] += arr[j];
                cout << ", " << i[j] << "\n";
            }
        }
    }
    return 0;
}

int main() {
    int arr[] = {10, -5};
    PointsShift(arr);
    return 0;
}

File: lecture_notes/10 refactoring/3-step.js
'use strict';

// Function that iterates each element in polyline checks if it's an obj; parse it using JSON.
const point_parse = (points) => {
    return typeof points != 'object' ? JSON.parse(points) : points
}

// Taking a result from PointParse and shifting it
const points_shift = (arr, dot_param) => {
    return arr.map((dot) => {
        dot['x'] += dot_param['x']
        dot['y'] += dot_param['y']
        return dot;
    });
}

const polyline = [
    { x: 0, y: 0 },
    { x: 10, y: 10 },
    '{ "x": 20, "y": 20 }',
    { x: 30, y: 30 },
];

// Iterate polyline's elements for PointParse and then put it as a parameter
const LineIter = polyline.map(point_parse)
console.log(points_shift(LineIter, { x: 10, y: -5 }))
